File: 0452_minimum_number_of_arrows/find_min_arrows.java
class Solution {
    // Time: O(n * log(n))
    // Space: O(1)
    public int findMinArrowShots(int[][] points) {
        if (points.length == 0) {
            return 0;
        }
        Arrays.sort(points, (a, b) -> Integer.compare(a[1], b[1]));
        int res = 1;
        int end = points[0][1];
        for (int[] point : points) {
            int x = point[0], y = point[1];
            if (x > end) {
                res++;
                end = y;
            }
        }
        return res;
    }
}

File: 0452_minimum_number_of_arrows/find_min_arrows.py
from typing import List


def findMinArrowShots(points: List[List[int]]) -> int:
    """
    Time: O(n * log(n))
    Space: O(1)
    """
    if not points:
        return 0
    res = 1
    points.sort(key=lambda x: x[1])
    end = points[0][1]
    for x, y in points:
        if x > end:
            res += 1
            end = y
    return res
